sklearn/model_selection/_search.py

```diff
@@ -793,14 +794,17 @@ def evaluate_candidates(candidate_params, cv=None, more_results=None):
                         "splits, got {}".format(n_splits, len(out) // n_candidates)
                     )
 
+                all_candidate_params.extend(candidate_params)
+                all_out.extend(out)
+                _warn_about_fit_failures(all_out, self.error_score)
+
                 # For callable self.scoring, the return type is only know after
                 # calling. If the return type is a dictionary, the error scores
                 # can now be inserted with the correct key. The type checking
                 # of out will be done in `_insert_error_scores`.
                 if callable(self.scoring):
-                    _insert_error_scores(out, self.error_score)
-                all_candidate_params.extend(candidate_params)
-                all_out.extend(out)
+                    _insert_error_scores(all_out, self.error_score)
```

Comment:
I am not sure about this `out` vs `all_out` thing now ... maybe I need to think about it a bit more ...

Reply:
OK thanks for the details, I used `out`